lib/srv/app/session.go

```diff
@@ -171,12 +172,25 @@ func (s *Server) withJWTTokenForwarder(ctx context.Context, sess *sessionChunk,
 		return trace.Wrap(err)
 	}
 
+	// Duplicate the default behavior for the header rewriter in oxy, as there
+	// is no getters/setters for this in the Forwarder so that we can grab it
+	// and re-set it there.
+	h, err := os.Hostname()
+	if err != nil {
+		h = "localhost"
+	}
```

Comment:
This looks out of place here. Can we just expose a `NewHeaderRewriter` constructor in oxy that would encapsulate this logic?

Reply:
Yes we can. I'll submit that PR shortly.